NineAnimator/Models/Anime Source/animehub.ac/AnimeHub+Episode(OLD).swift

```diff
@@ -0,0 +1,116 @@
+        "oserver": "Oserver",
+        "mpserver": "MPserver",
+        "yuserver": "YUserver"
+        //"hserver": "Hserver" Excluding until we update HydraX parser
+    ]
+
+    static let urlRegex = try! NSRegularExpression(pattern: #"((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=\+\$,\w]+@)?[A-Za-z0-9.-]+|(?:www.|[-;:&=\+\$,\w]+@)[A-Za-z0-9.-]+)((?:\/[\+~%\/.\w-_]*)?\??(?:[-\+=&;%@.\w_]*)#?(?:[\w]*))?)"#, options: .caseInsensitive)
+    
+    func episode(from link: EpisodeLink, with anime: Anime) -> NineAnimatorPromise<Episode> {
+        self.requestManager.request(
+            link.identifier,
+            handling: .browsing
+        ).responseString.thenPromise {
+            responseContent in
+            let bowl = try SwiftSoup.parse(responseContent)
+            
+            // Get available servers
+            let availableServerList = try bowl.select("#selectServer > option").map {
+                serverContainer in
+                try serverContainer.attr("sv")
+            } .filter {
+                // Exclude any servers that NineAnimator does not support
+                NASourceAnimeHub.knownServers.keys.contains($0)
+            }
+            
+            // Check if selected server is available
+            guard availableServerList.contains(link.server) else {
+                let alternativeEpisodeLinks = availableServerList.map {
+                    alternativeServer in EpisodeLink(
+                        identifier: link.identifier,
+                        name: link.name,
+                        server: alternativeServer,
+                        parent: link.parent
+                    )
+                }
+                // Throw EpisodeServerNotAvailableError with the list of alternatives
+                throw NineAnimatorError.EpisodeServerNotAvailableError(
+                    unavailableEpisode: link,
+                    alternativeEpisodes: alternativeEpisodeLinks,
+                    updatedServerMap: NASourceAnimeHub.knownServers
+                )
+            }
+            
+            // Extract Episode ID from EpisodeLink URL
+            let episodeURLParams = try URL(string: link.identifier, relativeTo: self.endpointURL).tryUnwrap()
+                .query
+                .tryUnwrap()
+            let episodeID = try formDecode(episodeURLParams).value(at: "ep", type: String.self)
+            
+            // Request api for episode iframe (if available)
+            return self.requestManager.request(
+                "ajax/anime/load_episodes_v2",
+                handling: .ajax,
+                query: ["s": link.server],
+                parameters: ["episode_id": episodeID]
+            ).responseData.then {
+                responseContent in
+                // Convert response into NSDictionary
+                let responseJSON = try JSONSerialization.jsonObject(with: responseContent, options: []) as! NSDictionary
```

Comment:
Avoid force type casting `as!`. Also, if the response is formatted as JSON, why don't you use a struct, and use `responseDecodable`?